src/components/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,162 @@
+
+  const handleMouseDown = () => {
+    setLastInputDevice('mouse');
+  };
+
+  const handleKeyDown = () => {
+    setLastInputDevice('keyboard');
+  };
+
+  type MoveThroughList = 'UP' | 'DOWN';
+
+  const handleSelectPrevOrNext = (action: MoveThroughList) => {
+    const selectedOptionIndex = optionList.indexOf(selectedOption);
+
+    if (selectedOptionIndex !== 0
+      && selectedOptionIndex !== optionList.length - 1) {
+      const nextIndex = action === 'UP'
+        ? selectedOptionIndex - 1
+        : selectedOptionIndex + 1;
+
+      setSelectedOption(optionList[nextIndex]);
+    }
+
+    if (selectedOptionIndex === 0) {
+      const nextIndex = action === 'UP'
+        ? optionList.length - 1
+        : selectedOptionIndex + 1;
+
+      setSelectedOption(optionList[nextIndex]);
+    }
+
+    if (selectedOptionIndex === optionList.length - 1) {
+      const nextIndex = action === 'UP'
+        ? selectedOptionIndex - 1
+        : 0;
+
+      setSelectedOption(optionList[nextIndex]);
+    }
+  };
+
+  const handleKeyBoardArrows = (
+    keyDownOnDropDownEvent: React.KeyboardEvent<HTMLDivElement>,
+  ) => {
+    switch (keyDownOnDropDownEvent.key) {
+      case 'ArrowUp':
+      case 'ArrowLeft':
+        return handleSelectPrevOrNext('UP');
+      case 'ArrowDown':
+      case 'ArrowRight':
+        return handleSelectPrevOrNext('DOWN');
+      default:
+        return null;
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, []); // to remove elements added on mount listener when it is unmounted
+
+  const handleSelectOption = (
+    optionClickEvent: React.MouseEvent<HTMLLIElement, MouseEvent>,
+  ) => {
+    optionClickEvent.preventDefault();
+
+    const dropDown = (optionClickEvent.target as HTMLLIElement)
+      .parentElement?.parentElement;
+
+    if (dropDown) {
+      dropDown.blur();
+    }
+
+    const selected = optionClickEvent.currentTarget.innerText;
+
+    setSelectedOption(selected);
+  };
+
+  const handleClickOnItSelf = (
+    clickOnMenu: React.MouseEvent<HTMLDivElement, MouseEvent>,
+  ) => {
+    clickOnMenu.preventDefault();
+    setLastInputDevice('mouse');
+
+    const dropDown = clickOnMenu.currentTarget;
+
+    if (isMenuOpened) {
+      const { currentTarget } = clickOnMenu;
+
+      currentTarget.blur();
+
+      return setIsMenuOpened(false);
+    } else {
+      dropDown.classList.add('clicked');
+    }
+
+    setIsMenuOpened((prevState) => !prevState);
+  };
+
+  const handleFocus = (
+    focusEvent: React.FocusEvent<HTMLDivElement, Element>,
+  ) => {
+    focusEvent.preventDefault();
+
+    const dropDown = focusEvent.currentTarget;
+
+    if (lastInputDevice === 'mouse') {
+      dropDown.classList.add('clicked');
+    } else if (lastInputDevice === 'keyboard') {
+      dropDown.classList.remove('clicked');
+    }
+  };
```

Comment:
You can add and remove classes on element with `className={classNames('clicked': isClicked)}`
So do it in `tsx`, not via `classList`